crates/indexer/src/models/token_models/v2_token_utils.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+
+use anyhow::{Context, Result};
+use aptos_api_types::deserialize_from_string;
+use bigdecimal::BigDecimal;
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Formatter};
+
+/// Tracks which token standard a token / collection is built upon
+#[derive(Serialize)]
+pub enum TokenStandard {
+    V1,
+    V2,
+}
+
+impl fmt::Display for TokenStandard {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        let res = match self {
+            TokenStandard::V1 => "v1",
+            TokenStandard::V2 => "v2",
+        };
+        write!(f, "{}", res)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ObjectCore {
+    pub allow_ungated_transfer: bool,
+    #[serde(deserialize_with = "deserialize_from_string")]
+    pub guid_creation_num: BigDecimal,
+    pub owner: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum V2TokenResource {
+    ObjectCore(ObjectCore),
+}
+
+impl V2TokenResource {
+    pub fn is_resource_supported(data_type: &str) -> bool {
```

Comment:
Haven't thought deeply enough about writing this code in a cleaner way tbh. Originally decided to do this to avoid converting it to MoveResources all the time, but this can likely be optimized.